### PR TITLE
Update LCD drivers

### LINKED ISSUE #238: ILI9341V LCD driver support (BSP-408)
Hi. I recently bought 2.8 IPS display module with ILI9341**V** chip (like this one http://www.lcdwiki.com/2.8inch_IPS_SPI_Module_ILI9341). The standard `esp_lcd_ili9341` driver refused to drive this display. So, I figured out that the difference is in initialization sequence. Here is the code snippet of `vendor_specific_init` array to make this module working. It will be good it this will be integrated in the `esp_lcd_ili9341` driver (probably through Kconfig with ability to select chip).

```
static const lcd_init_cmd_t vendor_specific_init[] = {
	/* SW reset */
	{LCD_CMD_SWRESET, {0}, 0},
	/* 200 ms delay */
	{0x80, {250}, 1},
	{0xCF, {0x00, 0xC1, 0X30}, 3},
	{0xED, {0x64, 0x03, 0X12, 0X81}, 4},
	{0xE8, {0x85, 0x00, 0x78}, 3},
	{0xCB, {0x39, 0x2C, 0x00, 0x34, 0x02}, 5},
	{0xF7, {0x20}, 1},
	{0xEA, {0x00, 0x00}, 2},
	{0xC0, {0x13}, 1},
	{0xC1, {0x13}, 1},
	{0xC5, {0x22, 0x35}, 2},
	{0xC7, {0xBD}, 1},
	{LCD_CMD_MADCTL, {(LCD_CMD_MV_BIT | 0x08)}, 1},
	{0x3A, {0x55}, 1},  //*** INTERFACE PIXEL FORMAT: 0x66 -> 18 bit; 0x55 -> 16 bit
	{0xF6, {0x01, 0x30}, 2},
	{0xB1, {0x00, 0x1B}, 2},
	{0xF2, {0x00}, 1},
	{0x26, {0x01}, 1},
	/* Positive gamma correction */
	{0xE0, {0x0F, 0x35, 0x31, 0x0B, 0x0E, 0x06, 0x49, 0XA7, 0x33, 0x07, 0x0F, 0x03, 0x0C, 0x0A, 0x00}, 15},
	/* Negative gamma correction */
	{0xE1, {0x00, 0x0A, 0x0F, 0x04, 0x11, 0x08, 0x36, 0x58, 0x4D, 0x07, 0x10, 0x0C, 0x32, 0x34, 0x0F}, 15},

	/* Column address set, SC=0, EC=0xEF */
	{LCD_CMD_CASET, {0x00, 0x00, 0x00, 0xEF}, 4},
	/* Page address set, SP=0, EP=0x013F */
	{LCD_CMD_RASET, {0x00, 0x00, 0x01, 0x3F}, 4},
	/* Memory write */
	{LCD_CMD_RAMWR, {0}, 0},
	/* Entry mode set, Low vol detect disabled, normal display */
	{0xB7, {0x07}, 1},
	/* Display function control */
	{0xB6, {0x08, 0x82, 0x27}, 3},

	/* Sleep out */
	{LCD_CMD_SLPOUT, {0}, 0x80},
	/* Invert colors */
	{LCD_CMD_INVON, {0}, 0},
	/* Display on */
	{LCD_CMD_DISPON, {0}, 0x80},

	{0, {0}, 0xff},
};
```

### PR DESCRIPTION
**Pull request by Lzw655**
# ESP-BSP Pull Request checklist

> Note: For new BSPs create a PR with this [link](https://github.com/espressif/esp-bsp/compare/main...my-branch?quick_pull=1&template=pr_template_bsp.md).

- [ ] Version of modified component bumped
- [ ] CI passing

# Change description
* Update GC9A01, GC9503, ILI9341, ST7796 
* Add LCD vendor config
* Output version in log
* Add test_apps to check build errors and memory leaking
* closes #238

### CODE REVIEW

**Comment by Lzw655** [comment]
Hi @espzav @tom-borcin, I have updated some familiar LCD drivers and validated them through test_apps on the hardware. As for the other LCD drivers, I haven't made any changes as I'm not familiar with them and don't have the hardware for validation.

PTAL, thank you!

**Comment by Lzw655** [comment]
Hi @espzav @tore-espressif, PTAL, thank you!

File: components/lcd/esp_lcd_st7796/CMakeLists.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-idf_component_register(SRCS "esp_lcd_st7796.c"
-                       INCLUDE_DIRS "include"
-                       REQUIRES "esp_lcd"
-                       PRIV_REQUIRES "driver")
+idf_component_register(SRCS "esp_lcd_st7796.c" INCLUDE_DIRS "include" REQUIRES "esp_lcd" PRIV_REQUIRES "driver")
```
[the remark this answers is not in the document]
Comment (Lzw655) [reply components/lcd/esp_lcd_st7796/CMakeLists.txt]: Sure.

File: components/lcd/esp_lcd_gc9a01/idf_component.yml [hunk 2]
```diff
@@ -1,5 +1,6 @@
-version: "1.0.1"
+version: "2.0.0"
```
[the remark this answers is not in the document]
Comment (Lzw655) [reply components/lcd/esp_lcd_gc9a01/idf_component.yml]: Ok.

File: components/lcd/esp_lcd_ili9341/idf_component.yml [hunk 3]
```diff
@@ -1,5 +1,6 @@
-version: "1.0.2"
+version: "2.0.0"
```
[the remark this answers is not in the document]
Comment (Lzw655) [reply components/lcd/esp_lcd_ili9341/idf_component.yml]: Ok.

**Review by tore-espressif — APPROVED** [review]
Changes LGTM, but I did not have a chance to check on HW.

@espzav can you please have a brief look?

**Comment by espressif2022** [comment]
@espzav ,PTAL. I plan to modify the screen init cmd of esp-box-3, which may be needed. Solved the problem of unclear screen display.

**Comment by espzav** [comment]
@Lzw655 Thank you for this PR. Great job! LGTM
I am so sorry for the delay. 
Note: Tested only HW GC9503 and ILI9341.

**Pull request merged**